server/routerlicious/packages/services-core/src/runWithRetry.ts

```diff
@@ -0,0 +1,41 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { delay } from "@fluidframework/common-utils";
+import { ILogger } from "./lambdas";
+
+export async function runWithRetry<T>(
+    api: () => Promise<T>,
+    callName: string,
+    maxRetries: number,
+    logger: ILogger,
```

Comment:
To account for logger being optional in lambdas, I think making this optional would be a good idea. Then just call it like `logger?.info(...)` to make sure we don't run into any undefined errors